From 80d2473f8103aebc041d901cfc699bd02dd4cbeb Mon Sep 17 00:00:00 2001
From: Sagi Maimon <smaimon@habana.ai>
Date: Thu, 28 Jan 2021 13:46:09 +0200
Subject: [PATCH] [SW-29734] preboot: security: support min fdt ver

Support loading of Uboot fit that has a min fdt version.
this version supports more strict parsing of fit image.

Change-Id: I7d28770b27e2fb247bb83bfaf0795bf2ed205dd8
Signed-off-by: Sagi Maimon <smaimon@habana.ai>
---

diff --git a/preboot/fit.c b/preboot/fit.c
index 09a5534..38602eb 100644
--- a/preboot/fit.c
+++ b/preboot/fit.c
@@ -51,6 +51,8 @@
 
 #define FIT_MAX_IMAGES		8
 
+#define FDT_MIN_SUPPORTED_VERSION 0x11
+
 /*
  * For checking decompressed image bounderies, assume the compression factor to
  * be 5 times bigger than the compressed data.
@@ -593,6 +595,10 @@
 	u64 load, start;
 	const char *cmpr;
 
+	/* Check if image supports a minimum FDT version or newer */
+	if (fdt_version(fit) < FDT_MIN_SUPPORTED_VERSION)
+		return FIT_LOAD_SEC_FAILED;
+
 	/* Get number of images in preboot configuration */
 	conf = fit_find_config_node(fit);
 	if (conf < 0)
